### PR TITLE
Added some more logging functionality

### LINKED ISSUE #4: Connect to API endpoints
**Description:** 
Connect to API endpoints 

**Outcome:**
Python file that connects to the API endpoints for the plant data

**User Story:**
As developers, we need to extract the needed information from the API endpoint so that we have data to transform and load for the visualisation.

**Design:**

- [x] API endpoints for (50 plants, plant 43 doesn't exist and plant 7 is on loan) (https://data-eng-plants-api.herokuapp.com/plants/<id>)
- [x] Load to csv file
- [x] Add logging

### PR DESCRIPTION
**Pull request by keoghrmj**
Some more logging has been added 

Closes #4

### CODE REVIEW

File: pipeline/Extract.py [hunk 1]
```diff
@@ -66,9 +66,9 @@ def fetch_and_collect_data() -> None:
             }
 
             collected_data.append(combined_data)
+            print(f"Processed plant ID: {number}")
         else:
-            print(f"Failed with status code: {response.status_code}")
-        print(f"Processed plant ID: {number}")
+            print(f"Failed with status code: {response.status_code}, plant ID: {number}")
```
Comment (SurinaCS) [inline pipeline/Extract.py:71]: beautiful stuff

File: pipeline/Extract.py [hunk 2]
```diff
@@ -79,6 +79,6 @@ def write_to_csv(data: list[dict], csv_file: str) -> None:
         writer.writerows(data)
 
 if __name__ == "__main__":
-    csv_file = "Api_information.csv"
+    csv_file = "Plant_information.csv"
```
Comment (SurinaCS) [inline pipeline/Extract.py:82]: clarity very good

**Review by FarihaChoudhury — CHANGES_REQUESTED** [review]
This is print statements not logging

File: pipeline/Extract.py [hunk 3]
```diff
@@ -66,9 +66,9 @@ def fetch_and_collect_data() -> None:
             }
 
             collected_data.append(combined_data)
+            print(f"Processed plant ID: {number}")
```
Comment (qrafiq1) [inline pipeline/Extract.py:69]: Use loggin instead of print statements?

**Review by qrafiq1 — CHANGES_REQUESTED** [review]

**Review by qrafiq1 — APPROVED** [review]

File: pipeline/Plant_information.csv [hunk 4]
```diff
@@ -1,48 +1,48 @@
...
+Eliza Andrews,eliza.andrews@lnhm.co.uk,(846)669-6651x75948,-13.7804,34.4587,Salima,MW,Medinilla Magnifica,Medinilla magnifica,https://perenual.com/storage/image/upgrade_access.jpg,13.288680852596213,37.18531383701657,2024-11-26 09:09:35,"Mon, 25 Nov 2024 14:12:39 GMT"
+Eliza Andrews,eliza.andrews@lnhm.co.uk,(846)669-6651x75948,37.49223,15.07041,Catania,IT,Calliandra Haematocephala,Calliandra haematocephala,https://perenual.com/storage/species_image/1477_calliandra_haematocephala/og/52063600268_834ebc0538_b.jpg,13.32643222871992,-1.08411359714286,2024-11-26 09:09:36,"Mon, 25 Nov 2024 13:30:18 GMT"
+Carl Linnaeus,carl.linnaeus@lnhm.co.uk,(146)994-1635x35992,14.14989,121.3152,Calauan,PH,Zamioculcas Zamiifolia,Zamioculcas zamiifolia,https://perenual.com/storage/image/upgrade_access.jpg,14.878464323084312,34.463350951171876,2024-11-26 09:09:38,"Mon, 25 Nov 2024 14:31:09 GMT"
+Eliza Andrews,eliza.andrews@lnhm.co.uk,(846)669-6651x75948,17.94979,-94.91386,Acayucan,MX,Crassula Ovata,Crassula ovata,https://perenual.com/storage/species_image/2193_crassula_ovata/og/33253726791_980c738a1e_b.jpg,9.354218607628031,33.09518671622935,2024-11-26 09:09:40,"Mon, 25 Nov 2024 14:02:15 GMT"
```
Comment (FarihaChoudhury) [inline pipeline/Plant_information.csv:48]: can we add this file to the .gitignore pls! can always run extract before doing transform

File: pipeline/Extract.py [hunk 5]
```diff
@@ -66,9 +75,9 @@ def fetch_and_collect_data() -> None:
             }
 
             collected_data.append(combined_data)
+            logging.info(f"Processed plant ID: {number}")
```
Comment (SurinaCS) [inline pipeline/Extract.py:78]: good use of logging

**Review by FarihaChoudhury — APPROVED** [review]
Looks good, thanks for implementing our comments

**Pull request merged**